proposals/function-pointers.md

```diff
@@ -263,62 +272,41 @@ We use `System.Runtime.InteropServices.OutAttribute`, applied as a `modreq` to t
-signature. That would be no worse than the support today.
-
-``` csharp
-void* v = &CloseHandle;
-delegate* cdecl<IntPtr, bool> f1 = (delegate* cdecl<IntPtr, bool>)v;
-```
-
-### Extensible set of unmanaged calling conventions
-
-The set of unmanaged calling conventions supported by the current ECMA-335 encodings is outdated. We have seen requests to add support
-for more unmanaged calling conventions, for example:
-
-- [vectorcall](https://docs.microsoft.com/cpp/cpp/vectorcall) https://github.com/dotnet/coreclr/issues/12120
-- StdCall with explicit this https://github.com/dotnet/coreclr/pull/23974#issuecomment-482991750
-
-The design of this feature should allow extending the set of unmanaged calling conventions as needed in future. The problems include
-limited space for encoding calling conventions (12 out of 16 values are taken in `IMAGE_CEE_CS_CALLCONV_MASK`) and number of places
-that need to be touched in order to add a new calling convention. A potential solution is to introduce a new encoding that represents
-the calling convention using [`System.Runtime.InteropServices.CallingConvention`](https://docs.microsoft.com/en-us/dotnet/api/system.runtime.interopservices.callingconvention) enum.
-
-For reference, https://github.com/llvm/llvm-project/blob/master/llvm/include/llvm/IR/CallingConv.h has the list of calling conventions
-supported by LLVM. While it is unlikely that .NET will ever need to support all of them, it demonstrates that the space of calling
-conventions is very rich.
+### Metadata Representation of Calling Conventions
+
+The following calling convention specifiers are specially recognized by the compiler:
+
+* `System.Runtime.CompilerServices.CallConvCdecl`
+* `System.Runtime.CompilerServices.CallConvStdcall`
+* `System.Runtime.CompilerServices.CallConvFastcall`
+* `System.Runtime.CompilerServices.CallConvThiscall`
+
+If the identifier specified for the `unmanaged` calling convention binds to one of these types, then the function pointer
+is emitted with the calling convention bit set to the appropriate value, and encodes the calling convention type as the first
+modopt on the return type of the function pointer. For any other `CallConv` type or combination of such types, the compiler
+emits a calling convention bit of Unmanaged (0x9), and encodes all `CallConv` types specified as the first modopts on the return
+type of the function pointer. What calling conventions and combinations are valid is determined by the runtime and platform of
+final execution, and no validation is performed that a valid combination of calling conventions has been specified beyond
+ensuring that all calling convention types come from the right namespace in the core library of the program. If the return
+type of a pointer is by `ref`, then all calling conventions will be encoded after the ref specifier.
+
+When interpreting a function pointer in metadata, all modopts that start with `CallConv` and come from the
```

Comment:
>all modopts [](start = 50, length = 11)

I think this doesn't quite match the latest proposal discussed on the internal thread.